onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -103,6 +186,13 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
     if (p_op_kernel == nullptr)
       return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Got nullptr from GetKernel for node: ",
                              node.Name());
+
+    std::string node_name = node.Name();
+    if (node_name.empty()) {
+      // Node name field is often blank in execution graph, so derive something meaningful for profile traces and logs.
+      node_name = node.OpType() + "_" + std::to_string(node_index);
+    }
```

Comment:
This line could potentially add significant cost to a small model. If we're going to do this, shouldn't we do it when loading the model so it's not a per-Run overhead?

Reply:
* Also, if node_name is not empty, you pay the cost of copying it in the previous line.
* Moreover, consider using += for constructing the string. Using + is probably fine here since the strings are small, but when new developers see this they might develop the habit of using + everywhere.

Reply:
per discussion with Sherlock, it's not trivial to do so and we might still miss it. So we move most of the logic to profiling blocks, except that the empty string will get recreated every time. The other way is to copy the same logic into 3 profiling blocks, but it looks a bit ugly.

---
In reply to: [397014455](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r397014455) [](ancestors = 397014455)